lib/ansible/vars/manager.py

```diff
@@ -525,16 +531,41 @@ def _get_delegated_vars(self, play, task, existing_variables):
         # as we're fetching vars before post_validate has been called on
         # the task that has been passed in
         vars_copy = existing_variables.copy()
+
+        # get search path for this task to pass to lookup plugins
+        vars_copy['ansible_search_path'] = task.get_search_path()
+
+        # ensure basedir is always in (dwim already searches here but we need to display it)
+        if self._loader.get_basedir() not in vars_copy['ansible_search_path']:
+            vars_copy['ansible_search_path'].append(self._loader.get_basedir())
+
         templar = Templar(loader=self._loader, variables=vars_copy)
 
         items = []
         has_loop = True
         if task.loop_with is not None:
             if task.loop_with in lookup_loader:
+                fail = True
+                if task.loop_with == 'first_found':
+                    # first_found loops are special. If the item is undefined then we want to fall through to the next
+                    fail = False
                 try:
                     loop_terms = listify_lookup_plugin_terms(terms=task.loop, templar=templar,
-                                                             loader=self._loader, fail_on_undefined=True, convert_bare=False)
-                    items = wrap_var(lookup_loader.get(task.loop_with, loader=self._loader, templar=templar).run(terms=loop_terms, variables=vars_copy))
+                                                             loader=self._loader, fail_on_undefined=fail, convert_bare=False)
+
+                    if not fail:
+                        loop_terms = [t for t in loop_terms if not templar.is_template(t)]
+
+                    mylookup = lookup_loader.get(task.loop_with, loader=self._loader, templar=templar)
+
+                    # give lookup task 'context' for subdir (mostly needed for first_found)
+                    for subdir in ['template', 'var', 'file']:  # TODO: move this to constants?
+                        if subdir in task.action:
+                            break
+                    setattr(mylookup, '_subdir', subdir + 's')
+
+                    items = wrap_var(mylookup.run(terms=loop_terms, variables=vars_copy))
```

Comment:
Should this pass wantlist=True to make sure it's a list?

Reply:
This is bypassing the `Templar._lookup` method, which I have other thoughts about but am not addressing here, and `wantlist` is handled by `Templar._lookup`.

Technically this function, as well as `TaskExecutor._get_loop_items` shares duplicated code with `Templar._lookup` as well...

Reply:
Oh, I see. It seems like we could remove the hardcoded wantlist=True from `_get_loop_items`. It's getting passed to the lookup there for some reason.